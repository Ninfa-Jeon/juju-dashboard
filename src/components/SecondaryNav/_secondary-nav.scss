@import "vanilla-framework/scss/settings_colors";

.p-secondary-nav {
  .p-list__link {
    color: $color-dark;
  }

  .p-list__item {
    padding-left: 1rem;

    &.is-selected {
      border-left: 3px solid $color-information;
      font-weight: 400;
      padding-left: calc(1rem - 3px);
    }
  }
}
