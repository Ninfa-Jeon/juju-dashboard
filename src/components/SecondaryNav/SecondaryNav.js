import React from "react";
import { Link } from "react-router-dom";
import classNames from "classnames";

import "./_secondary-nav.scss";

const pages = [
  { label: "Models", path: "/" },
  { label: "Clouds", path: "/clouds" },
  { label: "Kubernetes", path: "/kubernetes" },
  { label: "Controllers", path: "/controllers" },
  { label: "Usage", path: "/usage" },
  { label: "Logs", path: "/logs" }
];

const SecondaryNav = () => {
  const currentLocation = window.location.pathname;
  return (
    <nav className="p-secondary-nav">
      <ul className="p-list">
        {Object.values(pages).map(navItem => (
          <li
            key={navItem.path}
            className={classNames("p-list__item", {
              "is-selected": currentLocation === navItem.path
            })}
          >
            <Link className="p-list__link" to={navItem.path}>
              {navItem.label}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default SecondaryNav;
