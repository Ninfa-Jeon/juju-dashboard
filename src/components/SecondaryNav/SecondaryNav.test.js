import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { mount } from "enzyme";

import SecondaryNav from "./SecondaryNav";

describe("secondary Nav", () => {
  it("renders without crashing and matches snapshot", () => {
    const wrapper = mount(
      <Router>
        <SecondaryNav />
      </Router>
    );
    expect(wrapper).toMatchInlineSnapshot();
  });
});
