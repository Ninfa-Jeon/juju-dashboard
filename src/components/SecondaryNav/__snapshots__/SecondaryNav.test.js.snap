// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secondary Nav renders without crashing and matches snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SecondaryNav>
      <nav
        className="p-secondary-nav"
      >
        <ul
          className="p-list"
        >
          <li
            className="p-list__item is-selected"
            key="/"
          >
            <Link
              className="p-list__link"
              to="/"
            >
              <a
                className="p-list__link"
                href="/"
                onClick={[Function]}
              >
                Models
              </a>
            </Link>
          </li>
          <li
            className="p-list__item"
            key="/clouds"
          >
            <Link
              className="p-list__link"
              to="/clouds"
            >
              <a
                className="p-list__link"
                href="/clouds"
                onClick={[Function]}
              >
                Clouds
              </a>
            </Link>
          </li>
          <li
            className="p-list__item"
            key="/kubernetes"
          >
            <Link
              className="p-list__link"
              to="/kubernetes"
            >
              <a
                className="p-list__link"
                href="/kubernetes"
                onClick={[Function]}
              >
                Kubernetes
              </a>
            </Link>
          </li>
          <li
            className="p-list__item"
            key="/controllers"
          >
            <Link
              className="p-list__link"
              to="/controllers"
            >
              <a
                className="p-list__link"
                href="/controllers"
                onClick={[Function]}
              >
                Controllers
              </a>
            </Link>
          </li>
          <li
            className="p-list__item"
            key="/usage"
          >
            <Link
              className="p-list__link"
              to="/usage"
            >
              <a
                className="p-list__link"
                href="/usage"
                onClick={[Function]}
              >
                Usage
              </a>
            </Link>
          </li>
          <li
            className="p-list__item"
            key="/logs"
          >
            <Link
              className="p-list__link"
              to="/logs"
            >
              <a
                className="p-list__link"
                href="/logs"
                onClick={[Function]}
              >
                Logs
              </a>
            </Link>
          </li>
        </ul>
      </nav>
    </SecondaryNav>
  </Router>
</BrowserRouter>
`;
